hooks/useFetch.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect, useCallback } from "react";
+
+interface Options {
+  defaultValue?: any;
+  params?: Record<string, string>;
+}
+
+function useFetch(
+  url: string,
+  options: Options = {
+    defaultValue: null,
+    params: {},
+  }
+) {
+  const [data, setData] = useState(options.defaultValue);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchData = useCallback(async () => {
+    setIsLoading(true);
+    setError(null);
+    try {
+      const response = await fetch(
+        `${url}?${new URLSearchParams(options.params)}`
+      );
+      const data = await response.json();
+      if (response.ok) {
+        setData(data);
+      } else {
+        setError(data.msg);
+      }
+    } catch {
+      setError("There was an error, please try again.");
+    }
+    setIsLoading(false);
+    // eslint-disable-next-line
```

Comment:
**High priority**

We shouldn't add this line, find what is causing this.

Reply:
I disabled that line because in the previous project it showed me the warning as it appears in the image below. What I remember in the "Performance Session" that we had, they mentioned that when we used useCallback and useMemo it was better to pass just the properties instead of passing the entire object in the dependency array.... should we change the code so that this warning does not appear?

Reply:
![Screen Shot 2022-01-25 at 17 11 52](https://user-images.githubusercontent.com/73197053/151077555-855ddfe1-efcd-494b-95fe-26f15347ef3f.png)